components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -619,18 +622,50 @@ public void removeOAuthApplicationData(String consumerKey) throws IdentityOAuthA
     }
 
     /**
-     * Revoke authorization for OAuth apps by resource owners
+     * Get apps that are authorized by the given user
+     *
+     * @return OAuth applications authorized by the user that have tokens in ACTIVE or EXPIRED state
+     */
+    public OAuthConsumerAppDTO[] getAppsAuthorizedByUser() throws IdentityOAuthAdminException {
+
+        return getoAuthConsumerAppDTOS(PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername());
+    }
+
+    public OAuthRevocationResponseDTO revokeAuthorization(String username) throws IdentityOAuthAdminException {
+
+        OAuthConsumerAppDTO oAuthConsumerAppDTOs[] = getAuthorizedApps(username);
+        for (OAuthConsumerAppDTO oAuthConsumerAppDTO : oAuthConsumerAppDTOs) {
+            OAuthRevocationRequestDTO oAuthRevocationRequestDTO = new OAuthRevocationRequestDTO();
+            oAuthRevocationRequestDTO.setApps(new String[]{oAuthConsumerAppDTO.getApplicationName()});
+            oAuthRevocationRequestDTO.setAuthzUser(username);
+            oAuthRevocationRequestDTO.setConsumerKey(oAuthConsumerAppDTO.getOauthConsumerKey());
+            oAuthRevocationRequestDTO.setConsumerSecret(oAuthConsumerAppDTO.getOauthConsumerSecret());
+            if (log.isDebugEnabled()) {
+                log.debug("Revoking access tokens for Client ID: " + oAuthConsumerAppDTO.getOauthConsumerKey() +
+                        " & user: " + username);
+            }
+            return revokeAuthzForApps(oAuthRevocationRequestDTO);
```

Comment:
is this return statement location is correct? or do we need the for loop?

Reply:
+1 need to fix